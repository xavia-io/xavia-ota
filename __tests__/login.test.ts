import { createMocks } from 'node-mocks-http';

import loginEndpoint from '../pages/api/login';

describe('Login API', () => {
  const originalEnv = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...originalEnv };
  });

  afterAll(() => {
    process.env = originalEnv;
  });

  it('should return 405 for non-POST requests', async () => {
    const { req, res } = createMocks({ method: 'GET' });
    await loginEndpoint(req, res);
    expect(res._getStatusCode()).toBe(405);
    expect(JSON.parse(res._getData())).toMatchSnapshot();
  });

  it('should return 500 if admin password is not configured', async () => {
    const { req, res } = createMocks({
      method: 'POST',
      body: { password: 'test' },
    });

    process.env.ADMIN_PASSWORD = '';
    await loginEndpoint(req, res);
    expect(res._getStatusCode()).toBe(500);
    expect(JSON.parse(res._getData())).toMatchSnapshot();
  });

  it('should return 401 for invalid password', async () => {
    const { req, res } = createMocks({
      method: 'POST',
      body: { password: 'wrong' },
    });

    process.env.ADMIN_PASSWORD = 'correct';
    await loginEndpoint(req, res);
    expect(res._getStatusCode()).toBe(401);
    expect(JSON.parse(res._getData())).toMatchSnapshot();
  });

  it('should return 200 for correct password', async () => {
    const { req, res } = createMocks({
      method: 'POST',
      body: { password: 'correct' },
    });

    process.env.ADMIN_PASSWORD = 'correct';
    await loginEndpoint(req, res);
    expect(res._getStatusCode()).toBe(200);
    expect(JSON.parse(res._getData())).toMatchSnapshot();
  });
});
