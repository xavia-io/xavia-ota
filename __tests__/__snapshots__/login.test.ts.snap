// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login API should return 200 for correct password 1`] = `
Object {
  "success": true,
}
`;

exports[`Login API should return 401 for invalid password 1`] = `
Object {
  "error": "Invalid password",
}
`;

exports[`Login API should return 405 for non-POST requests 1`] = `
Object {
  "error": "Method not allowed",
}
`;

exports[`Login API should return 500 if admin password is not configured 1`] = `
Object {
  "error": "Admin password not configured",
}
`;
